dj_hetmech_app/views.py

```diff
@@ -17,3 +20,51 @@ class NodeView(ModelViewSet):
     serializer_class = NodeSerializer
     filter_backends = (filters.SearchFilter,)
     search_fields = ('identifier', 'metanode__identifier', 'name')
+
+
+class QueryPairView(APIView):
+    http_method_names = ['get']
+
+    def polish_pathcounts(self, source_id, target_id, pathcounts_data):
+        """Polish pathcounts data to make source/target consistent with
+        user queries.
+        """
+
+        for entry in pathcounts_data:
+            # If necessary, swap "source_degree" and "target_degree"
+            # values in each pathcount entry's "dgp".
+            reversed = int(source_id) != entry['source']
+            entry['reversed'] = reversed
+            if reversed:
+                entry['dgp']['source_degree'], entry['dgp']['target_degree'] = (
+                    entry['dgp']['target_degree'], entry['dgp']['source_degree']
+                )
+            # Delete 'source' and 'target' fields.
+            del entry['source']
+            del entry['target']
+        return pathcounts_data
+
+    def get(self, request):
+        source_id = request.query_params.get('source', None)
+        target_id = request.query_params.get('target', None)
+        try:
+            source_node = Node.objects.get(pk=source_id)
+            target_node = Node.objects.get(pk=target_id)
+        except:
+            raise Http404
+
+        path_counts = PathCount.objects.filter(
+            Q(source=source_id, target=target_id) |
+            Q(source=target_id, target=source_id)
+        )
+
+        data = {}
+        data['source'] = NodeSerializer(source_node).data
+        data['target'] = NodeSerializer(target_node).data
+        data['pathCounts'] = self.polish_pathcounts(
```

Comment:
Can we switch `pathCounts` to `path_counts`? For our data fields, let's try to use underscore separation rather than camel case for consistency.

Reply:
Sure. (I used camelCase because this API will be parsed by JavaScript, which prefers that style.)